File: app/src/main/java/com/android/nsystem/testapps/LoginContract.java
package com.android.nsystem.testapps;

public interface LoginContract {

    interface View {
        void notifyLoginValid();
        void clearLoginInput();
        void enableLoginButton();
        void disableLoginButton();
    }

    interface Presenter {
        boolean isLoginInputValid(Account account);
        void submitLogin(Account account);
        void loginInputTextChanged(Account account);
    }
}

File: app/src/main/java/com/android/nsystem/testapps/LoginContract.kt
package com.android.nsystem.testapps

/**
 * @author Putra Nugraha (putra.nugraha@dana.id)
 * @version LoginContract, v 0.1 13/12/20 23.35 by Putra Nugraha
 */
interface LoginContract {

    interface View {

        fun notifyLoginValid()

        fun enableLoginButton()

        fun disableLoginButton()

        fun showProgress()

        fun dismissProgress()
    }

    interface Presenter {

        fun isLoginInputValid(account: Account): Boolean

        fun submitLogin(account: Account)

        fun loginInputTextChanged(account: Account)
    }
}